tests/test_io.py

```diff
@@ -73,13 +73,18 @@ def test_deserialize(message, message_parameters_dict, caplog):
             assert test_model.missing_schema
 
 
-def test_stream_read(circular_msg, circular_text):
-    with patch("hop.io._Consumer", MagicMock()) as mock_consumer:
-        mock_consumer.stream.return_value = [{'hey', 'you'}]
+def test_stream_read(circular_msg, circular_text, mock_broker, mock_consumer):
+    topic = "gcn"
+    group_id = "test-group"
+    start_at = io.StartPosition.EARLIEST
 
-        broker_url1 = "kafka://hostname:port/gcn"
-        broker_url2 = "kafka://hey@hostname:port/gcn"
-        start_at = io.StartPosition.EARLIEST
+    mock_adc_consumer = mock_consumer(mock_broker, topic, group_id, start_at)
+    with patch("hop.io.consumer.Consumer", autospec=True) as mock_instance:
+        mock_instance.side_effect = mock_adc_consumer
+        mock_adc_consumer.stream.return_value = [{'hey', 'you'}]
+
+        broker_url1 = "kafka://hostname:port/{topic}"
+        broker_url2 = "kafka://{group_id}@hostname:port/{topic}"
```

Comment:
Should these be `f`-strings?

Reply:
Definitely. This is now fixed, nice catch.